resources/compass/templates/agent-configuration-job.yaml

```diff
@@ -0,0 +1,104 @@
+    name: compass-agent-configuration
+    namespace: compass-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: compass-agent-configuration
+  namespace: compass-system
+rules:
+  - apiGroups: [""]
+    resources: [configmaps]
+    verbs: [create, list, get, watch]
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: compass-agent-configuration
+  namespace: compass-system
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  backoffLimit: 0
+  template:
+    metadata:
+      name: compass-agent-configuration
+      annotations:
+        sidecar.istio.io/inject: "true"
+    spec:
+      shareProcessNamespace: true
+      serviceAccountName: compass-agent-configuration
+      restartPolicy: Never
+      containers:
+        - name: compass-agent-configuration
+          image: "linkyard/kubectl:1.14.2"
+          command:
+            - bash
+            - -c
+            - |
+              set -e
+              MAX_RETRIES=60
+              DIRECTOR_URL=http://{{ .Values.global.gateway.host }}.{{ .Release.Namespace }}.svc.cluster.local:3000/director/graphql
+
+              function checkIfDirectorReady() {
+                  local cnt=0
+                  set +o errexit
+
+                  while :
+                  do
+                    directorRes=$(curl -s "${DIRECTOR_URL}" \
+                      --write-out "%{http_code}\n" --output /dev/null \
+                      -H 'Content-Type: application/json' \
+                      -H 'tenant: {{ .Values.global.defaultTenant }}' \
+                      --data-binary '{
+                          "query":"query queryRuntimes{\n  result: runtimes {\n    data {\n      id\n    }\n  }\n}"
+                      }')
+                    if [[ "$directorRes" == "200" ]]
+                      then
+                        echo "Director ready."
+                        break
+                      else
+                        ((cnt++))
+                        if (( cnt > $MAX_RETRIES )); then
+                          echo "Max retries has been reached (retries $MAX_RETRIES). Exit."
+                          exit 1
+                        fi
+
+                        echo "Director not ready! StatusCode: '${directorRes}' - waiting 5s..."
+                        sleep 5
+                      fi
+                  done
+
+                  set -o errexit
+              }
+              checkIfDirectorReady
+
+              export RUNTIME_ID=`curl "${DIRECTOR_URL}" \
+                  -H 'Content-Type: application/json' \
+                  -H 'tenant: {{ .Values.global.defaultTenant }}' \
+                  --data-binary '{
+                    "query":"mutation createRuntime {\n  result: createRuntime(\n    in: { name: \"kymaruntime\", description: \"Default Kyma runtime\" }\n  ) {\n    id\n  }\n}\n"
+                  }' | jq '.data .result .id'`
```

Comment:
What if there will be a different status code than 200? The script should print error and exit

Reply:
done, when jq checks a string, in case of failure the whole job will fail